### PR TITLE
make 'yarn start' be aware of BUILD_OUT_ROOT

### PR DESCRIPTION
**Pull request by peterbe**
Fixes https://github.com/mdn/content/pull/6349

Now, if you first run:

```
yarn start
``` 
in one terminal, and
```
yarn build
```
in another. Now both the `yarn build` and the `yarn start` will share the same overwritten `BUILD_OUT_ROOT` environment variable.

### CODE REVIEW

**Comment by peterbe** [comment]
@hamishwillee If you can test this, I'll merge it.

**Comment by peterbe** [comment]
...unless @fiji-flo gets a chance to review it before you.

**Comment by hamishwillee** [comment]
I'm hoping @fiji-flo will get to it first. The queue from being away for 2 weeks is out of control.

**Comment by fiji-flo** [comment]
@peterbe  Looks fine and working but in what scenario does `yarn start` build something?

**Comment by hamishwillee** [comment]
Question (that should not block this going in). What's the best way to make this visible to potential users? The per-page flaw checker is visible on the top of page when you edit a page locally, but how would you find this trick?

**Comment by peterbe** [comment]
> Question (that should not block this going in). What's the best way to make this visible to potential users? The per-page flaw checker is visible on the top of page when you edit a page locally, but how would you find this trick?

There's a link to the "Flaws Dashboard" on the (Writer's) home page.

**Comment by hamishwillee** [comment]
So is this now tested?

**Comment by peterbe** [comment]
> So is this now tested?

What do you mean? 

If you're asking about this PR; what it does is that it makes it possible to start the "Flaws Dashboard" and get a complete picture of ALL flaws when you build ALL content.

**Comment by peterbe** [comment]
@hamishwillee To speed up the review of this; can you manually apply this change that this PR does to your local `package.json` and then do the two separate terminals as described above. 
If you're now able to get a complete picture of all flaws with filtering and sorting etc, then let is know that it fully works.

**Comment by hamishwillee** [comment]
Hi @peterbe 
Sadly, not working for me. 
1. On Windows, synced to your branch, ran `yarn install` then `yarn start`. As you can see below this is the version with correct BUILD_OUT_ROOT
   image
2. Other CMD prompt ran `yarn build`

I can see it building, but the dashboard at `http://localhost:5000/en-US/_flaws` is showing zero issues after quite a few minutes.

**Comment by peterbe** [comment]
> P.S. What I meant previously about testing was "Is it enough to say I see flaws appearing in the list". If not, what do you need tested about this.

That is correct. I'd be very surprised if there are 0 flaws :)

**Comment by peterbe** [comment]
> Sadly, not working for me.

Not for me either! 
Something must have regressed.

**Comment by peterbe** [comment]
> > Sadly, not working for me.
> 
> Not for me either!
> Something must have regressed.

Now I know what it was! 
To solve this whole problem, it required/requires 2 separate solutions:

1. A fix to mdn/yari
2. A fix to mdn/content

The latter one is *this* PR. And the fix to `mdn/yari` was: https://github.com/mdn/yari/pull/4093

So what I did was merged in the latest `main` into this PR. So when you do `yarn install` the version of `node_modules/@mdn/yari/server/flaws.js` works. 

In conclusion, @hamishwillee please try again. Check out my latest branch and run `yarn install` again before you run `yarn start`.

**Comment by hamishwillee** [comment]
It works. A thing of beauty. Get it in :-).

For next iteration, consider providing a visual hint that the page is regenerating when reordering the table. It takes a while on my computer, during which nothing appears to happen. Not a problem except I thought that it was broken.

**Comment by peterbe** [comment]
I'm going to merge it then. If it works, it works.
